proxies/live/apiproxy/resources/jsc/validate-client-assertion-claims.js

```diff
@@ -0,0 +1,26 @@
+
+function extractJsonVariable(contextVariableName) {
+  return JSON.parse(
+    context.getVariable(
+      "jwt.DecodeJWT.FromClientAssertionFormParam."
+      + contextVariableName
+    )
+  )
+}
+
+
+jwtHeaders = extractJsonVariable('header-json')
+jwtPayload = extractJsonVariable('payload-json')
+
+// Set context variables based on the condition in the second argument
+context.setVariable('InvalidJwt.MissingKidHeader', !jwtHeaders.kid);
+context.setVariable('InvalidJwt.MissingOrInvalidTypHeader', typeof jwtHeaders.typ != "string" || jwtHeaders.typ.toLowerCase() != "jwt");
+context.setVariable('InvalidJwt.MissingExpClaim', !jwtPayload.exp);
+context.setVariable('InvalidJwt.InvalidExpiryTime', typeof jwtPayload.exp != "number");
+context.setVariable('InvalidJwt.MissingOrInvalidIssClaim', !jwtPayload.iss || jwtPayload.iss != jwtPayload.sub);
+context.setVariable('InvalidJwt.MissingJtiClaim', !jwtPayload.jti);
+context.setVariable('InvalidJwt.InvalidJtiClaim', typeof jwtPayload.jti != "string");
```

Comment:
what if it is an empty string or an invalid jti?

Reply:
Empty string will raise the fault in the line above, and any other string is valid according to the spec - which states only that it should be a unique string